### DISCUSSION TITLE
Speeding up GNU-diff

### DISCUSSION THREAD

**Discussion by gvanem** [BurntSushi/ripgrep#2101] [General] [answered]
I have a big desire to speed-up a recursive GNU-diff using some techniques used in RigGrep.
A `diff --recursive dir1 dir2` can be pitiful slow.
Like using:
 * Use memory-mapped files.
 * Split the job in multiple threads.
 * Read each directory branch by accessing the MFT directly?

But I'm a *n00b* when it comes to Rust (so I'll use plain C).
Is there anybody with Rust knowledge have some good ideas to achieve this?

**Comment by BurntSushi** [comment, marked as answer]
I think this question is a bit too broad for ripgrep Q&A. If you have more specific questions about techniques ripgrep uses, then I'd be happy to try and field them, but as it stands you're basically asking me to design an implementation for a totally different tool.

**Reply by gvanem** [reply]
>  you're basically asking me to design an implementation for a totally different tool.

No. I was just fishing for some ideas.
